pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,241 @@
+
+	var pkgInfo struct {
+		Name    string `json:"name"`
+		Version string `json:"version"`
+	}
+
+	file, err := os.ReadFile(path + "/package.json")
+	if err != nil {
+		return fmt.Errorf("error opening file: %w", err)
+	}
+	err = json.Unmarshal(file, &pkgInfo)
+
+	// Determine which tag to set
+	// npm adds `latest` as the default tag, and we want that to mean the newest released version.
+	npmTag := tag
+	if npmTag == "" {
+		switch {
+		case strings.Contains(pkgInfo.Version, "-alpha"):
+			npmTag = "dev"
+		case strings.Contains(pkgInfo.Version, "dev"):
+			npmTag = "dev"
+		case strings.Contains(pkgInfo.Version, "-beta"):
+			npmTag = "beta"
+		case strings.Contains(pkgInfo.Version, "-rc"):
+			npmTag = "rc"
+		default:
+			npmTag = "latest"
+		}
+	}
+
+	pkgNameWithVersion := pkgInfo.Name + "@" + pkgInfo.Version
+
+	// Verify version doesn't already exist
+	infoCmd := exec.Command(ex, "info", pkgNameWithVersion)
+	fmt.Printf("Running %s\n", infoCmd)
+	output, err := infoCmd.Output()
+	if err != nil {
+		return fmt.Errorf("could not run npm info on package")
+	}
+
+	if string(output) == "" {
+		fmt.Println("The version does not exist yet, and it is safe to publish")
+		fmt.Printf("Publishing %s to npm package registry...\n", pkg)
+		npmPublishCmd := exec.Command(ex, "publish", path, "-tag", npmTag)
+		publishOutput, publishErr := npmPublishCmd.CombinedOutput()
+
+		fmt.Print(string(publishOutput))
+
+		// handle errors
+		if publishErr != nil {
+			fmt.Println("error publishing package, retrying...")
+			// first, check if TOC/TOU issue
+			infoCheckCmd := exec.Command("npm", "info", pkgNameWithVersion)
+			checkOutput, checkErr := infoCheckCmd.Output()
+			if err != nil {
+				return fmt.Errorf("running npm info to verify failed %w", checkErr)
+			}
+			if string(checkOutput) == "" {
+				// this means the package was not published. We can be helpful to the user here.
+				if strings.Contains(string(publishOutput), "npm ERR! code EOTP") {
+					fmt.Printf("You have 2FA set up on your npm registry, " +
+						"which will cause publishing to fail due to an interactive OTP prompt.\n" +
+						"Please set up your `.npmrc` file to use a scoped or automation token.\n" +
+						"Read more about access tokens in the npm docs:\n" +
+						"https://docs.npmjs.com/about-access-tokens \n")
+				}
+
+				return fmt.Errorf(" %w", publishErr)
+			}
+		}
+		fmt.Println("success! published to npm")
+		return nil
+
+	}
+	return fmt.Errorf("did not publish %s to npm package registry, likely because the version already exists\n", pkg)
+}
+
+func publishToPyPI(pkg string) {
+	// TODO: implement me
+	fmt.Printf("Publishing %s to PyPI package registry...\n", pkg)
+}
+
+func publishToMaven(pkg string) {
+	// TODO: implement me
+	fmt.Printf("Publishing %s to Maven package registry...\n", pkg)
+}
+
+func publishToNuget(pkg string) {
+	// TODO: implement me
+	fmt.Printf("Publishing %s to nuget package registry...\n", pkg)
+}
```

Comment:
If the initial version we intend to land won't have these implemented, I think it would be preferable not to define these at all and mark them as unsupported languages until these are implemented.